config/src/errors.rs

```diff
@@ -0,0 +1,180 @@
+	UnsupportedSchema(String),
+
+	// Key derivation error
+	#[error("Key derivation error: {0}")]
+	KeyDerivationError(String),
+
+	// page_id: {}, prids len should be equal to connections len (connections: {}, prids: {})
+	#[error(
+		"page_id: {0}, prids len should be equal to connections len (connections: {1}, prids: {2})"
+	)]
+	PridsLenShouldBeEqualToConnectionsLen(String, usize, usize),
+
+	// generic error
+	#[error("Unknown error caught: {0}")]
+	UnknownError(String),
+
+	// Invalid private schema id
+	#[error("Invalid private schema id")]
+	InvalidPrivateSchemaId,
+
+	// Graph is full error
+	#[error("Graph is full")]
+	GraphIsFull,
+
+	// GraphState instance is full
+	#[error("GraphState instance is full")]
+	GraphStateIsFull,
+
+	// No resolved active key found error
+	#[error("No resolved active key found")]
+	NoResolvedActiveKeyFound,
+
+	// New page for existing page id error
+	#[error("New page for existing page id")]
+	NewPageForExistingPageId,
+
+	// Failed to retrieve new graph page error
+	#[error("Failed to retrieve graph page")]
+	FailedToRetrieveGraphPage,
+
+	// Duplicate connection detected error
+	#[error("Add of duplicate connection in another page detected")]
+	DuplicateConnectionDetected,
+
+	// Connection not found error
+	#[error("Connection not found")]
+	ConnectionNotFound,
+
+	// Duplicate update events error
+	#[error("Duplicate update events detected")]
+	DuplicateUpdateEvents,
+
+	// Call to private friends in public graph error
+	#[error("Call to private friends in non private graph")]
+	CallToPrivateFriendsInPublicGraph,
+
+	// Call to PRIDs in public graph error
+	#[error("Calling apply_prids in non private friendship graph!")]
+	CallToPridsInPublicGraph,
+
+	// Event exists
+	#[error("Event exists")]
+	EventExists,
+
+	// Invalid key
+	#[error("Invalid public key")]
+	InvalidPublicKey,
+
+	// Invalid secret key
+	#[error("Invalid secret key")]
+	InvalidSecretKey,
+
+	// Public key not compatible with secret key
+	#[error("Public key not compatible with secret key")]
+	PublicKeyNotCompatibleWithSecretKey,
+
+	// Unable to decrypt private graph with any of the imported keys
+	#[error("Unable to decrypt private graph with any of the imported keys")]
+	UnableToDecryptGraphChunkWithAnyKey,
+
+	// Incompatible privacy type for blob export
+	#[error("Incompatible privacy type for blob export")]
+	IncompatiblePrivacyTypeForBlobExport,
+
+	// Failed to acquire write lock on state manager
+	#[error("Failed to acquire write lock on state manager")]
+	FailedtoWriteLockStateManager,
+
+	// Failed to acquire read lock on state manager
+	#[error("Failed to acquire read lock on state manager")]
+	FailedtoReadLockStateManager,
+}
+
+impl From<apache_avro::Error> for DsnpGraphError {
+	fn from(error: apache_avro::Error) -> Self {
+		// Convert the underlying error to DsnpGraphError variant
+		// based on your error handling logic
+		// For example:
+		DsnpGraphError::AvroError(error.to_string())
+	}
+}
```

Comment:
Why not instead define `DsnpGraphError::AvroError` as:
```
#[error(transparent)]
AvroError(#[from] apache_avro::Error)
```

...will auto-generate a `From<apache_avro::Error>` implementation

Reply:
Yep i like that